Tools/unicode/makeunicodedata.py

```diff
@@ -207,12 +208,13 @@ def makeunicodedata(unicode, trace):
                     comp_first[l] = 1
                     comp_last[r] = 1
                     comp_pairs.append((l,r,char))
-                try:
-                    i = decomp_data.index(decomp)
-                except ValueError:
+                key = tuple(decomp)
+                i = decomp_data_cache.get(key, -1)
+                if i == -1:
                     i = len(decomp_data)
                     decomp_data.extend(decomp)
                     decomp_size = decomp_size + len(decomp) * 2
+                    decomp_data_cache[key] = i
```

Comment:
For the case of a cache hit, can we perhaps add an assertion to ensure everything is fine:
```py
                assert decomp == decomp_data[i:i+len(decomp)]
```

Reply:
done, thanks